File: creational/abstract-factory-example.py
class Condo:
    pass


class House:
    pass


class Hotel:
    pass


class Shop:
    pass


class BuildingFactory:
    @staticmethod
    def get_hotel():
        return Hotel()

    @staticmethod
    def get_shop():
        return Shop()

    @staticmethod
    def get_house():
        return House()

    @staticmethod
    def get_condo():
        return Condo()

    @staticmethod
    def get_building(typ):
        return globals()[typ]() if typ in globals().keys() else None


if __name__ == "__main__":
    user_request = ['Condo', 'Shop']
    user_properties = [BuildingFactory.get_building(t) for t in user_request]
    print(user_properties)

    user_properties = [Condo(), Shop()]
    print(user_properties)

File: creational/abstract-factory-example.js
let constructors = {};

// ----------------------------------------------------------------------
class Condo {
    building = "This is Condo";

    get() {
        return this.building;
    }
}
constructors.Condo = Condo;

// ----------------------------------------------------------------------
class House {
    building = "This is House";

    get() {
        return this.building;
    }
}
constructors.House = House;

// ----------------------------------------------------------------------
class Hotel {
    building = "This is Hotel";

    get() {
        return this.building;
    }
}
constructors.Hotel = Hotel;

// ----------------------------------------------------------------------
class Shop {
    building = "This is Shop";

    get() {
        return this.building;
    }
}
constructors.Shop = Shop;

// ----------------------------------------------------------------------
class BuildingFactory {
    static getCondo() {
        return new Condo();
    }

    static getHouse() {
        return new House();
    }

    static getHotel() {
        return new Hotel();

    }

    static getShop() {
        return new Shop();
    }

    static getBuilding(type) {
        if (Object.keys(constructors).indexOf(type) === -1) {
            throw new Error(`Our factory doesn't carry this type: ${type}!`);
        } else {
            return new constructors[type]();
        }
    }
}

// ----------------------------------------------------------------------
const userRequest = ['Condo', 'Shop'];
for (let i in userRequest) {
    console.log(BuildingFactory.getBuilding(userRequest[i]).get());
}

console.log('- - - - - - - -');
let user_properties = [new Condo(), new Shop()];
console.log(user_properties);

console.log('- - - - - - - -');
let newRequest = BuildingFactory.getHotel().get();
console.log(newRequest);